src/test/java/dal/fatih/todoly/TodolyTest.java

```diff
@@ -1,252 +1,553 @@
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+        , properties = "spring.profiles.active=test")
 public class TodolyTest {
 
-    private final Connection connection = new DBConnection().getConnection();
-    private ByteArrayOutputStream outContent;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-    public TodolyTest() throws SQLException {
-    }
+    @LocalServerPort
+    private int randomServerPort;
 
-    @Before
-    public void setUp() {
-        outContent = new ByteArrayOutputStream();
-        System.setOut(new PrintStream(outContent));
-    }
+    @Autowired
+    private TestRestTemplate testRestTemplate;
 
-    @After
-    public void tearDown() {
-        try {
-            Statement statement = connection.createStatement();
-            statement.executeUpdate("TRUNCATE TABLE task");
-            connection.close();
-            statement.close();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
+    @Autowired
+    private TaskRepository taskRepository;
 
     @Test
-    public void shouldNotAllowWrongSelection() {
-        provideInput(Arrays.asList("10", "q"));
-        App.main(new String[]{});
-        Assert.assertTrue(outContent.toString().contains("Invalid input"));
+    public void shouldCreateTask() throws URISyntaxException {
+        URI taskCreateUrl = new URI(createURLWithPort("/task"));
+        HttpEntity<TaskDTO> request = new HttpEntity<TaskDTO>(
+                new TaskDTO("Created-title-of-task", "Created-description-of-task"
+                        , LocalDateTime.now().plusDays(10))
+        );
+
+        ResponseEntity<String> responseEntity = this.testRestTemplate
+                .postForEntity(taskCreateUrl, request, String.class);
+        logger.warn(responseEntity.getBody());
+
+        String response = responseEntity.getBody();
+        assert response != null;
+        long createdTaskId = Long.parseLong(response.substring(6, response.length() - 1));
+
+        String actual = taskRepository.get(createdTaskId).getTitle();
+        String expected = "Created-title-of-task";
+
+        assertThat(actual, is(equalTo(expected)));

```

Comment:
If the task doesn't exist, you will likely get a `NullPointerException` which is bad. You should only make sure if the task object is not null. Then you can check each field in it to see if it matches what you intended to create.

```suggestion
        Task actual = taskRepository.get(createdTaskId);

        assertThat(actual, notNullValue());
        assertThat(actual.getTitle(), equalTo(taskDto.getTitle()));
        assertThat(actual.getDescription(), equalTo(taskDto.getDescription()));
        assertThat(actual.getDueDate(), equalTo(taskDto.getDueDate()));

```